sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -50,6 +50,10 @@ object GpuProjectExec extends Arm {
         val result = expr.columnarEval(cb)
         result match {
           case cv: ColumnVector => cv
+          case scalar: Scalar =>
```

Comment:
The same code to handle the `expr.columnarEval` result keeps popping up, making me think it would be worth factoring out this boilerplate code into something in `GpuExpressionUtils` that can be easily reused.  This can be a followup issue rather than tackled here if desired.

Reply:
Good suggestion, the followup issue is here https://github.com/NVIDIA/spark-rapids/issues/2327